pyccl/halos/profiles_cib.py

```diff
@@ -189,21 +189,19 @@ def _Lumcen(self, M, a):
         return Lumcen
 
     def _Lumsat(self, M, a):
-        Lumsat = np.zeros_like(M)
-        # Loop over Mparent
-        # TODO: if this is too slow we could move it to C
-        # and parallelize
-        for iM, Mparent in enumerate(M):
-            if Mparent > self.Mmin:
-                # Array of Msubs (log-spaced with 10 samples per dex)
-                nm = max(2, int(np.log10(Mparent/1E10)*10))
-                msub = np.geomspace(1E10, Mparent, nm+1)
-                # Sample integrand
-                dnsubdlnm = self.dNsub_dlnM_TinkerWetzel10(msub, Mparent)
-                Lum = self._Lum(np.log10(msub), a)
-                integ = dnsubdlnm*Lum
-                Lumsat[iM] = simps(integ, x=np.log(msub))
-        return Lumsat
+        res = np.zeros_like(M)
+        M_use = M[M >= self.Mmin, None]
+        logM = np.log10(M_use)
+        LOGM_MIN = 10
+        nm = max(2, 10*int(np.max(logM) - LOGM_MIN))
+        msub = np.linspace(LOGM_MIN, np.max(logM), nm+1)[None, :]
```

Comment:
I think you probably need to check that `np.max(M) > self.Mmin`, and if that's not the case, just return 0 everywhere.

Reply:
ok done